### ISSUE TITLE
Time to live property on container

### ISSUE THREAD

**Issue by ViswanathVeerappan** [microsoft/spring-data-cosmosdb#523]
Hi Team, 
We are using spring data cosmos db for one of the applications in our company. The problem we are noticing is with the annotation @Document with default values. Having the default values enables the application to create the container if it does not exist with ru set to 4000 and collection name as the model object name except the time to live property on the container. This is by default set to "Off" on the container even though the application sets a value of -1. Is this an expected behavior wherein we have to enable it at the container level on the azure portal?

**Comment by kushagraThapar** [comment]
@ViswanathVeerappan - Yes, that is expected behavior. `-1` value of default time to live indicates that items within that container will never expire. If you want to change the value, you can do that through annotation itself. Also, if you want to change any of the values, you can do that through `@Document` annotation. 
Here is some documentation on default time to live values : https://azuresdkdocs.blob.core.windows.net/$web/java/azure-cosmos/3.7.0/com/azure/data/cosmos/CosmosContainerProperties.html#defaultTimeToLive(java.lang.Integer)

**Comment by ViswanathVeerappan** [comment]
Thanks for your response! [https://docs.microsoft.com/en-us/azure/cosmos-db/how-to-time-to-live#java3-enable-noexpiry], as per the documentation mentioned there highlighted by these points
- When DefaultTimeToLive is null then your Time to Live is Off
- When DefaultTimeToLive is -1 then your Time to Live setting is On (No default)
- When DefaultTimeToLive has any other Int value (except 0) your Time to Live setting is On
We do not see On(No default) setting enabled rather it is Off by default even for the value -1. 

Our idea is to have the container enabled with On(No default) setting and have document or item level ttl enabled to delete the items.

**Comment by kushagraThapar** [comment]
@ViswanathVeerappan  - So your ask is to enable time to live on items and use default value on the container level ?
As far as I know, v3 Cosmos SDK (on which spring-data-cosmosdb SDK is based) doesn't support setting time to live values for individual items. It only supports setting it on container. 
Let me search for a workaround if exists any.

**Comment by ViswanathVeerappan** [comment]
Thanks for your response! Following are the cases we tried,

1) We tried to set the time to live property present in the @Document annotation to a value greater than zero. The other properties had default values set for them. It created the container with the default values enabled(for ru, collection name and auto created it when it was not present) except the ttl on the container. In order for the ttl to work, the container should have either -1 or a value greater than 0. But the container created had the property turned off.
2) Similarly, in the other case, We had the defaults enabled in the @Document annotation. The container created had the same behavior. It did not have the time to live enabled.

The containers created in both the cases should have had the ttl enabled. To be more specific, in the first case it should have been "On" and in the second case, "On(no default)". We were not noticing this behavior.

The only time the ttl worked was when we had the time to live property set to "On(no default)" on the container using the Azure portal and had a field called ttl for each of the items being inserted.(https://docs.microsoft.com/en-us/azure/cosmos-db/how-to-time-to-live#portal-set-ttl-item). Our requirement is to have configurable time to live value which can be modified on the fly(more from a document perspective). So On(no default) on the container and item level ttl enabled.

To summarize, containers auto created from the application with a positive value or a default value(-1) set for time to live within in the @Document annotation are not having the time to live enabled on the container. We are looking to see, if we can get some help from you.

**Comment by kushagraThapar** [comment]
@ViswanathVeerappan  - Here is the PR for the above bug fix: https://github.com/microsoft/spring-data-cosmosdb/pull/524

**Comment by kushagraThapar** [comment]
@ViswanathVeerappan - Would release v2.2.3.FIX2 work for you ?

**Comment by ViswanathVeerappan** [comment]
@kushagraThapar - is new patch version possible?

**Comment by kushagraThapar** [comment]
@ViswanathVeerappan  - do you mean v2.2.4 ?
Its just naming, so I don't mind either of them.

**Comment by ViswanathVeerappan** [comment]
@kushagraThapar - yeah 2.2.4. would prefer this.

**Comment by kushagraThapar** [comment]
@ViswanathVeerappan - Sounds good, I will release this in next couple of days and will close this issue. Will update you over here.

**Comment by ViswanathVeerappan** [comment]
@kushagraThapar - Thanks for the quick response! Really appreciate your help.

**Comment by kushagraThapar** [comment]
This has been fixed and released: https://github.com/microsoft/spring-data-cosmosdb/releases/tag/v2.2.4

**Comment by kushagraThapar** [comment]
@ViswanathVeerappan  - Please upgrade and confirm the fix.

**Comment by anfeldma-ms** [comment]
Hi @ViswanathVeerappan any update on upgrading and confirming the fix? Thanks.

**Comment by ViswanathVeerappan** [comment]
@kushagraThapar @anfeldma-ms - We tested the fix made and it is able to configure the time to live property when the collection gets created. Thanks for the help!

**Comment by anfeldma-ms** [comment]
@ViswanathVeerappan thanks!

**Comment by yiliuTo** [comment]
Hi, is the timeToLive in @Document on container level or item level?

**Comment by kushagraThapar** [comment]
@yiliuTo it is on container level.

**Comment by yiliuTo** [comment]
@kushagraThapar  Got it, thanks.
